src/python/bot/tasks/corpus_pruning_task.py

```diff
@@ -768,11 +777,25 @@ def _process_corpus_crashes(context, result):
     task_creation.create_tasks(testcase)
 
 
-def _get_cross_pollinate_fuzzers(engine_name, current_fuzzer_name):
-  """Return a list of fuzzer objects to use for cross pollination."""
-  cross_pollinate_fuzzers = []
-
+def _select_targets_and_jobs_for_pollination(engine_name, current_fuzzer_name,
+                                             method, tag):
+  """Select jobs to use for cross pollination."""
   target_jobs = list(fuzz_target_utils.get_fuzz_target_jobs(engine=engine_name))
+
+  if method == TAGGED_METHOD:
+    similar_tagged_targets = [
+        target for target in corpus_tagging.get_targets_with_tag(tag)
+        if target.fully_qualified_fuzz_target_name != current_fuzzer_name
+    ]
+    # Intersect target_jobs and similar_tagged_targets on fully qualified
+    # fuzz target name.
+    target_jobs = [
+        target for target in target_jobs if target.fuzz_target_name in [
+            tagged.fully_qualified_fuzz_target_name
```

Comment:
let's put these in a separate list or set, otherwise it'll be unnecessarily iterating, plus that code gets hard to read

Reply:
Done